### PR TITLE
Support ORC file read by writer time zone

### PR DESCRIPTION
**Pull request by KevinyhZou**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Support ORC file read by writer time zone

Sometimes, we write a orc format file (which has a `datatime64` type field) with local timezone , like `Asia/Shanghai`, but when we read from it,  it would always use `GMT` timezone to read the `datetime64` field by default, which would lead to the wrong result, different from the write value. 

In my case, when write a `datetime64` field by `Asia/Shanghai` timezone, like `2024-06-30 20:00:00` into a orc file, but the value read from it would be `2024-07-01 04:00:00`.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit 9341bbe19733fc2da455c6b0c51d0ee9bb13bf49 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/66025/9341bbe19733fc2da455c6b0c51d0ee9bb13bf49/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Review by Avogar — APPROVED** [review]

File: src/Core/Settings.h [hunk 1]
```diff
@@ -1046,6 +1046,7 @@ class IColumn;
     M(Bool, input_format_orc_allow_missing_columns, true, "Allow missing columns while reading ORC input formats", 0) \
     M(Bool, input_format_orc_use_fast_decoder, true, "Use a faster ORC decoder implementation.", 0) \
     M(Bool, input_format_orc_filter_push_down, true, "When reading ORC files, skip whole stripes or row groups based on the WHERE/PREWHERE expressions, min/max statistics or bloom filter in the ORC metadata.", 0) \
+    M(Bool, input_format_orc_read_use_writer_time_zone, false, "Whether use the writer's time zone in ORC stripe for ORC row reader, the default ORC row reader's time zone is GMT.", 0) \
```
Comment (den-crane) [inline src/Core/Settings.h:1049]: GMT ? Or UTC ? GMT!=UTC

Comment (KevinyhZou) [reply src/Core/Settings.h:1049]: It is `GMT`, and this is defined at `orc/c++/src/Options.hh`, in `struct RowReaderOptionsPrivate`.

Comment (KevinyhZou) [reply src/Core/Settings.h:1049]: It is `GMT`, and this is defined at `orc/c++/src/Options.hh`, in `struct RowReaderOptionsPrivate`. @den-crane

**Pull request merged**